### ISSUE TITLE
Can´t add Hue wall switch to Zigbee group

### ISSUE THREAD

**Issue by thmnxo4** [Koenkk/zigbee2mqtt#19941]
### What happened?

Adding switch to group fails

### Zigbee2MQTT version

1.33.2

### Adapter firmware version

6.10.3.0 build 297

### Adapter

EZSP v8

### Debug log

Error 2023-12-01 07:38:14Failed to add from group (Command 0x001788010b007ad4/1 genGroups.add({"groupid":1,"groupname":""}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 34097 - 1 - 134 - 4 - 0 after 10000ms))

**Comment by chrisvandiemen** [comment]
Experiencing the same problem.

**zigbee2mqtt version**: 1.34.0
**adapter**: SONOFF ZigBee 3.0 USB Dongle Plus, TI CC2652P
**debug log**: Failed to add from group (Command 0x001788010d9fb29c/1 genGroups.add({"groupid":1,"groupname":""}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 22193 - 1 - 214 - 4 - 0 after 10000ms))

**Comment by jamesonuk** [comment]
Have you toggle the switch before trying to add it to the group?   You need to wake it up before trying to add it to a group (I normally press it a few times whilst Z2M is trying to send the join group command just to make sure)

**Comment by chrisvandiemen** [comment]
Thanks for the help!

Yep, did try that. Toggling before, during or after the add command doesn’t
help. I did pair the Hue Wall via a router. Not sure if that is relevant
but is worth trying pairing directly to the coordinator.

Op do 14 dec 2023 om 12:08 schreef jameson_uk ***@***.***>

**Comment by thmnxo4** [comment]
Yes, I did so

jameson_uk ***@***.***> schrieb am Do., 14. Dez. 2023, 12:08:

**Comment by snurre0** [comment]
I'm having the same issue with my wall switch. I'm following the guide for directly controlling zigbee groups from here: https://www.zigbee2mqtt.io/devices/929003017102.html

But step 2 fails when trying to add it to a group, and I get the same error as above:
Error 2024-01-13 18:23:09Failed to add from group (Command 0x001788010b031100/1 genGroups.add({"groupid":2,"groupname":""}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 15042 - 1 - 114 - 4 - 0 after 10000ms))

The thing is that after continuing with the binding, I get the wallswitch to control my lights, but in a super weird way.
When I press my switch really quickly it works, but holding it for more than a split second makes the light go wither on or off again, depending on if I turned them off or on. Holding the button will dim them fully either way, but releasing the button makes them turn either on or off, depending on the state they're coming from.

Anyone else experiencing this with the wall switch? Is this related to the issue where I can't add the wallswitch to the group?

**Comment by thmnxo4** [comment]
Hi, will this be fixed in the near future?

**Comment by thmnxo4** [comment]
Seems to be fixed. Tried again with 1.37 and group binding worked.

**Comment by mwerder** [comment]
Got the same issue, following the guide but cannot add the module to the group

**Comment by SkyhawkXava** [comment]
Same here. Can't add to a group and binding doesn't work.

z2m version: 1.42.0-2

zigbee-model: RDM004
firmware-Date: 20220510
firmware-Version: 2.60.16

Btw: This device is not listet in the ota section.

**Comment by SkyhawkXava** [comment]
I have now found a solution: First, “Groups” must be bound to the coordinator. After that, the required steps can be carried out.

But: Dimming (hold button) is not working. (Push Button is activated.)

**Comment by fkuiper** [comment]
I've got the same issue with my switches. I create a zigee group and adding lights works perfectly, adding the switch will give:

`error: 	z2m: Failed to add from group (ZCL command 0x001788010cc574da/1 genGroups.add({"groupid":4,"groupname":""}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"reservedBits":0,"writeUndiv":false}) failed (Timeout - 42216 - 1 - 178 - 4 - 0 after 10000ms))`

z2m version: 1.42.0
coordinator: slzb-06p7 running firmware: 20240716 (radio) v2.3.6 (core)
switch firmware: 1.0.5

Triggering the switch makes no difference. I've added this same switch and light to a group successfully on my previous coordinator (a Conbee II USB dongle).

**Comment by MfCrizz** [comment]
I have the same problem with my switches. I cannot add them to the group. However, I am wondering if this is necessary because if I skip this step in the instructions, I can still add the switch to the on/off cluster of the group. 

The switch can then also turn the lights on and off, but then I have the problem that if I turn the group on or off in a different way than with the switch (e.g. sending a mqtt message), I have to press the switch twice to control the group again.

**Comment by hatti75** [comment]
Same problem here.

**Comment by fkuiper** [comment]
After a lot of try-and-error I found out that:

- You do need to add the switch to the group! Trigger the switch right after you added it to the group, but it will still give a time-out error. You can ignore this error and you won't see it in the UI but some how something is changed somewhere (sorry, I don't have any idea what or where).
- After this you need to unbind the OnOff Cluster from the coordinator. Of course you need to trigger the switch right after you click unbind so it wakes up.
- Now you can bind the OnOff Cluster to the group (again you need to trigger to wake up the switch).

For me this worked like a charm. My groups now toggle on and off without any automation in home assistant; it works for both lights as well as smart plugs; and it even works if zigbee2mqtt or home assistant is down (this will greatly improve your wife acceptance factor!). If your lights support dimming, you can also unbind and bind manuSpecificPhilips together with OnOff so the lights will dimm up or down if you hold a puls switch for longer than a second.

If you still whish to trigger some automation when you switch the lights on or off you can do on of two things:

1. Use the device as tirgger and make sure you select the device that is coupled to the group. I noticed the light does not always get triggered somehow (it does switch on or off thou).
2. You can rebind the OnOff Cluster to the coordinator (make sure this is the last step!) and trigger on the switch device again... Make sure the device action is 'toggle'. I had to do this in the yml-code because home assistant UI didn't detect the change in actions correctly.


IMO the time-out after you add the switch to the group and it not showing as a group member is a bug in zigbee2mqtt and is related to the zStack. I didn't have these issues when I used my Conbee II stick and somehow in the zigbee devices themself it does get registered correctly if you follow these steps.

**Comment by hatti75** [comment]
The binding works indeed. But if the bound lamp is triggered by something else, I have to push the switch twice.

**Comment by SkyhawkXava** [comment]
> After a lot of try-and-error I found out that:
> 
> * You do need to add the switch to the group! Trigger the switch right after you added it to the group, but it will still give a time-out error. You can ignore this error and you won't see it in the UI but some how something is changed somewhere (sorry, I don't have any idea what or where).
> * After this you need to unbind the OnOff Cluster from the coordinator. Of course you need to trigger the switch right after you click unbind so it wakes up.
> * Now you can bind the OnOff Cluster to the group (again you need to trigger to wake up the switch).
> 
> For me this worked like a charm. My groups now toggle on and off without any automation in home assistant; it works for both lights as well as smart plugs; and it even works if zigbee2mqtt or home assistant is down (this will greatly improve your wife acceptance factor!). If your lights support dimming, you can also unbind and bind manuSpecificPhilips together with OnOff so the lights will dimm up or down if you hold a puls switch for longer than a second.
> 
> If you still whish to trigger some automation when you switch the lights on or off you can do on of two things:
> 
> 1. Use the device as tirgger and make sure you select the device that is coupled to the group. I noticed the light does not always get triggered somehow (it does switch on or off thou).
> 2. You can rebind the OnOff Cluster to the coordinator (make sure this is the last step!) and trigger on the switch device again... Make sure the device action is 'toggle'. I had to do this in the yml-code because home assistant UI didn't detect the change in actions correctly.
> 
> IMO the time-out after you add the switch to the group and it not showing as a group member is a bug in zigbee2mqtt and is related to the zStack. I didn't have these issues when I used my Conbee II stick and somehow in the zigbee devices themself it does get registered correctly if you follow these steps.

Which zigbee-model? RDM001 or RDM004?

**Comment by SkyhawkXava** [comment]
> The binding works indeed. But if the bound lamp is triggered by something else, I have to push the switch twice.

The RDM001 and RDM005 behave differently, as far as I have been able to research:

**RDM001:**
	•	Group binding is possible.
	•	The switch sends a “_toggle_” command to the device.
	•	Dimming works on long press if the “Level Control” cluster has been bound to the group.

**RDM004:**
	•	Group binding is possible, but only if the “Groups” cluster has also been bound to the coordinator.
	•	The switch sends “_on/off_” commands to the device.
	•	Dimming does not work on long press, even if the “Level Control” cluster has been bound to the group.

Since the RDM004 version only sends “on/off” commands and not the “toggle” command, the behavior described by you, hatti75, is correct. The Hue wall switch internally stores the last status (on or off). If the group is switched by another switch, the Hue wall switch does not receive this information.

As a battery-powered switch, it quickly returns to standby mode after each activation, during which it no longer receives messages or actively listens. As a result, the last stored group status in the Hue wall switch may be outdated. This can lead to situations where, for example, the Hue wall switch sends an “on” command even though the group has already been turned on by another switch.

**Comment by fkuiper** [comment]
I have them both in my network and didn't notice any difference between them. In fact, I didn't know I've had two different versions up until now :)

I also use automations to switch on and off lights which I can also toggle using the switches. I've never noticed any odd behavior like needing to toggle my switches twice after an automation changed the state. I only modify via the groups, not an individual light btw, maybe that is something?

Do you use pulse or rocker switches? Mine are all pulse...

-------- Oorspronkelijk bericht --------
Op 23-02-2025 22:03, schreef SkyhawkXava :

**Comment by SkyhawkXava** [comment]
I modify only via groups, as well. I use only push switches.

**Comment by hatti75** [comment]
Ist doesn‘t work for me. I tried different bindings to the coordinator (with/without Groups cluster). I anyway get the error message when adding it to the group. I tried it at least 20 times. Binding to the group works, but both switches do not toogle. They send on/off commands to the bound group and so the command can be wrong if trigged by other sources.

z2m version: 2.1.1
Coordinator type: zStack3x0
Coordinator revision: 20210708

zigbee-model 1: RDM004
firmware-Date: 20240618
firmware-Version: 2.77.35

zigbee-model 2: RDM001
firmware-Date: 20240618
firmware-Version: 2.77.35

**Comment by fkuiper** [comment]
I've got two versions running:

RDM001 - 20210504
RDM004 - 20220510

I've never noticed any difference between them. As I said before: up until now I didn't even know they were different:).

Verzonden met Proton Mail Android

-------- Oorspronkelijk bericht --------
Op 23-02-2025 23:03, schreef SkyhawkXava :

**Comment by Pferdebockwurst** [comment]
I can confirm the description at the zigbee2mqtt devices [page](https://www.zigbee2mqtt.io/devices/929003017102.html#directly-control-zigbee-groups) works perfectly fine for two of my switches, which are both RDM001, on of them being on firmware 1.0.3 and one on 1.0.5. Will try my RDM004 later.

**Comment by hatti75** [comment]
Seems to be a problem with newer firmware... Mine are on the latest.

**Comment by fkuiper** [comment]
Hmm... maybe the never switch firmware has a bug in it. Mine is pretty old I see now. Apparently the OTA doesn't work on my switches... which could maybe be a good thing :)

-------- Oorspronkelijk bericht --------
Op 24-02-2025 19:01, schreef hatti75 :

**Comment by Pferdebockwurst** [comment]
@Koenkk if you want to investigate this, I could send you one of these switches with a recent firmware. I am very interested in getting them to work when bound to a group too.

**Comment by Pferdebockwurst** [comment]
I have found [this](https://github.com/dresden-elektronik/deconz-rest-plugin/issues/7947) discussion and indeed the RDM004 is reported to behave very different. But is this really dependent on the new firmware?

**Comment by SkyhawkXava** [comment]
I do think it’s due to the newer firmware version. @hatti75 has the latest 2024 firmware version on both revisions and obviously, both revisions (RDM001 and RDM004) have the issues.

**Comment by Pferdebockwurst** [comment]
So then the RDM001 must have been manually updated by him. Because as far as I know the RDM001 has never been shipped with a new firmware 2.x.x.
But where does this firmware file come from and how did he update?

**Comment by Pferdebockwurst** [comment]
> I've got two versions running:
> 
> RDM001 - 20210504
> RDM004 - 20220510
> 
> I've never noticed any difference between them. As I said before: up until now I didn't even know they were different:).
> 
> Verzonden met Proton Mail Android
> […](#)

Are you sure the RDM004 works the same way as the RDM001? Because I have both devices, too, with the same firmware. While the RDM001 works perfectly as described, I can confirm that the RDM004 only sends on/off instead of toggle and doesn't support dimming.

**Comment by eXitus01** [comment]
Started Zigbee2MQTT from scratch an get the same error:

```
z2m: Failed to add from group (ZCL
command ox001788010cc25314/1
genGroups.add({"groupid":1,"groupname"
esponse":true,"direction":0,"reservedBits
":O,"writeUndiv":false}) failed
({"target":65459,"apsFrame":
profileld " : 260," clusterld " :4," sourceEndp
quence":218,"commandldentifier":0}
timed out after 10000ms))
```

zigbee2mqtt version: 2.1.1-
adapter: Nabu Casa Home Assistant SkyConnect ZBT-1 (Ember)

switch type: push button
wallswitch modell: RDM001
wallswitch firmware: 2.77.35 (20240618)

**Comment by hatti75** [comment]
> So then the RDM001 must have been manually updated by him. Because as far as I know the RDM001 has never been shipped with a new firmware 2.x.x. But where does this firmware file come from and how did he update?

The switches were both paired to the original HUE bridge before I paired them with zigbee2mqtt. The HUE bridge must have updated them.

**Comment by Pferdebockwurst** [comment]
> Just throwing this in here that I had 10 of these that were all the older ones (I guess RDM001) on firmware version 1.0.5 / 20210504 (or some even older ones on 1.0.3 / 20210115).
> 
> I have just ordered another two and the one that turned up is on firmware 2.60.16 / 20220510 (Model on the back says 9290030171A which doesn't quite match the one linked above) so kooks like the `A` the RDM004 and `02` the RDM001. (Using Dev Console to read `ModelId` from `genBasic` cluster on new and old switches does seem to confirm that.
> 
> With my Conbee II I was able to pair the device and add it to group without issue. It has gone a bit funny when doing the binding so I wonder if there is something about the ordering of events
> 
> If people are having issues adding it to group then I guess we might want to gather what coordinator people are using and what model is printed on the back (or using the dev console to query the model being reported).

The problem is especially that the newer model variant and the old variant that has been upgraded to the newer firmware won't send toogle commands but on/off instead.

**Comment by jamesonuk** [comment]
> The problem is especially that the newer model variant and the old variant that has been upgraded to the newer firmware won't send toogle commands but on/off instead.

? most of the comments above are about not being able to add to group (as in title).
Isn't the toggle vs on/off a case of setting the right device mode? (rocker = on/off, push button = toggle) ?

**Comment by fkuiper** [comment]
> > The problem is especially that the newer model variant and the old variant that has been upgraded to the newer firmware won't send toogle commands but on/off instead.
> 
> ? most of the comments above are about not being able to add to group (as in title). Isn't the toggle vs on/off a case of setting the right device mode? (rocker = on/off, push button = toggle) ?

I think so... all my switches send the toggle message. I did notice that in some cases I had to reconfigure them to push button after joining them to the group and/or binding their end points (don't remember which one it was, it's been a couple of months ;)) even though the UI already showed it was configured as such (just switched between rocker and push button to make sure it's configured correctly).

**Comment by Pferdebockwurst** [comment]
No, I cannot confirm that the newer revision or the updated old revision sends the toogle command, even when configured als push button. But to be sure, I will try again.

**Comment by jamesonuk** [comment]
> No, I cannot confirm that the newer revision or the updated old revision sends the toggle command, even when configured als push button. But to be sure, I will try again.

No I think you are right, I am not sure what I was thinking there 😕 ....
I don't see how it could do anything other than toggle.   If you switch on the light (group) via HA and then flick the switch it should turn it off.   Press it again and it should turn on.   This has been the way they have always worked for me.   If they were to maintain state on the switch and send explicit on/off then if you turned on the lights via HA then flicking the switch would do nothing as it would send an `on` command as it would think the lights are off and then if you flick it again it will send an `off` command which would work.     It has to send a toggle for it to make sense no?

**Comment by Pferdebockwurst** [comment]
> > No, I cannot confirm that the newer revision or the updated old revision sends the toggle command, even when configured als push button. But to be sure, I will try again.
> 
> No I think you are right, I am not sure what I was thinking there 😕 .... I don't see how it could do anything other than toggle. If you switch on the light (group) via HA and then flick the switch it should turn it off. Press it again and it should turn on. This has been the way they have always worked for me. If they were to maintain state on the switch and send explicit on/off then if you turned on the lights via HA then flicking the switch would do nothing as it would send an `on` command as it would think the lights are off and then if you flick it again it will send an `off` command which would work. It has to send a toggle for it to make sense no?

Yes, and that's exactly the problem. The newer firmware seems to remember the last command that had been sent. Under normal circumstances on/off would work the same, but not if the lamp had been toggled otherwise (for example by an automation).

**Comment by jamesonuk** [comment]
I have created https://github.com/Koenkk/zigbee2mqtt/issues/26775 to continue the discussion about RDM001 vs RDM004 / 1.x vs 2.x firmware.

We should continue discussion there about the differences and what is / isn't working and leave this thread as adding device to group as per OP

**Comment by marithpl** [comment]
I found a solution that worked for me yesterday after a few hours of struggling. 

I went into the Hue Wall Module's settings and disconnected it from the coordinator. then added it to a group and bind it to that group. Then I re-bound the Wall Hue Module to the coordinator.

**Comment by jamesonuk** [comment]
> I went into the Hue Wall Module's settings and disconnected it from the coordinator. then added it to a group and bind it to that group. Then I re-bound the Wall Hue Module to the coordinator.

Was this with a 1.x or 2.x firmware?

**Comment by marithpl** [comment]
Firmware ID
1.0.5

I bind this device a year Ago. But it was factory reset - after that i had a problem with binding. 

Above stepa helped me
